Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -822,7 +819,28 @@ protected class ClassWriter : IClassWriter {
 
     protected override ConcreteSyntaxTree EmitForStmt(IToken tok, IVariable loopIndex, bool goingUp, string endVarName,
       List<Statement> body, LList<Label> labels, ConcreteSyntaxTree wr) {
-      throw new UnsupportedFeatureException(tok, Feature.ForLoops);
+      var iterator = endVarName != null ? "range" : "count";
+      wr.Write($"for {IdName(loopIndex)} in {iterator}(");
+      var startWr = wr.Fork();
+      if (!goingUp) {
+        wr.Write("-1");
+      }
+      if (endVarName != null) {
+        wr.Write($", {endVarName}");
+        if (!goingUp) {
+          wr.Write("-1");
+        }
+      }
+      if (!goingUp) {
+        wr.Write(", -1");
+      }
```

Comment:
IO think this would be more readable with the four cases spelled out separately; WDYT?